code/WorkInProgress/missile_latejoin.dm

```diff
@@ -40,41 +42,70 @@
 	//	..()
 
 	// its fucking lunch time
-	proc/lunch(var/mob/living/person, var/d = NORTH)
+	proc/lunch(mob/living/person, d=null)
 		if (!(person))
 			return 0
 		passenger = person
 		person.set_loc(src)
-		move_dir = d
-		dir = d
+		if(d)
+			src.set_dir(d)
 		passenger << 'sound/effects/bamf.ogg'
-		//playsound(src.loc, "sound/effects/bamf.ogg", 100, 0)
 		sleep(0.1 SECONDS)
 		passenger << 'sound/effects/flameswoosh.ogg'
 		//playsound(src.loc, "sound/effects/flameswoosh.ogg", 100, 0)
 		ion_trail.start()
 		move_self()
 
+	proc/set_dir(d)
+		move_dir = d
+		dir = d
+		src.transform = null
+		if(d & (WEST | EAST))
+			var/matrix/tr = src.transform.Turn(dir2angle(d))
+			src.transform = tr.Translate(16, -16)
+		if(d & (NORTH | SOUTH))
+			src.bound_width = 32
+			src.bound_height = 64
+			src.ion_trail.yoffset = 13
+		else
+			src.bound_width = 64
+			src.bound_height = 32
+			src.ion_trail.yoffset = 0
+
 	proc/move_self()
 		var/glide = 0
 
-		while (moved_on_flooring <= 4)
+		while (moved_on_flooring <= MISSILE_FLOORS_TO_STOP)
 			glide = (32 / MISSILE_SPAWN_MOVEDELAY) * world.tick_lag// * (world.tick_lag / CLIENTSIDE_TICK_LAG_SMOOTH)
 			src.glide_size = glide
 			src.animate_movement = SLIDE_STEPS
 			passenger.glide_size = glide
 			passenger.animate_movement = SYNC_STEPS
-			//step(src,NORTH)
-			src.y++
+			src.loc = get_step(src, src.move_dir) // I think this is supposed to be loc= and not set_loc, not sure
+			if(!src.loc)
+				src.reset_to_random_pos()
 			src.glide_size = glide
 			src.animate_movement = SLIDE_STEPS
 			passenger.glide_size = glide
 			passenger.animate_movement = SYNC_STEPS
 			sleep(MISSILE_SPAWN_MOVEDELAY)
 
-			if (istype(get_turf(src), /turf/simulated/floor) && get_area(src))
-				moved_on_flooring++
+			var/area/AR = get_area(src)
+			var/turf/T = get_turf(src)
+			if (istype(T, /turf/simulated/floor) && !AR.teleport_blocked && !T.density)
```

Comment:
Your check prevents them from hitting the syndicate outpost since it's teleport blocked, but I believe they could still end up in various other spots like the ghost drone factory, mechanics outpost, and other on station Z random debris locations

perhaps we should verify that the area is of type `area/station` to help avoid that

Reply:
Hmmm. Probably for the best. Though I wouldn't mind people landing in the ghost drone factory.